src/common/components/repository-row/index.js

```diff
@@ -347,14 +347,20 @@ class RepositoryRow extends Component {
             TODO: show 'Loading' when waiting for status?
               and also show 'Never built' when no builds available
           */}
-          { latestBuild &&
+          { latestBuild && snap.snapcraft_data &&
             <BuildStatus
               link={ `/${fullName}/builds/${latestBuild.buildId}`}
-              status={ latestBuild.status }
+              colour={ latestBuild.status }
               statusMessage={ latestBuild.statusMessage }
               dateStarted={ latestBuild.dateStarted }
             />
           }
+          { !snap.snapcraft_data &&
```

Comment:
It would be better to have both of the conditions joined together to be sure the result is always one `<BuildStatus>`. 
When we have them separate like this it can lead to mistake when we change the condition on only on of them.
If the conditional is too complex to keep in JSX maybe it's worth extracting it to a variable or function the will return `BuildStatus` depending on the state.